bokeh/core/property/container.py

```diff
@@ -186,8 +186,13 @@ def from_json(self, json, *, models=None):
     def validate(self, value, detail=True):
         super().validate(value, detail)
 
+        if isinstance(value, dict):
+            for key, val in value.items():
+                value[key] = self.values_type.transform(val)
```

Comment:
This is clearly a wrong place to put this code. Validation should not modify the input in anyway.